src/wp-includes/option.php

```diff
@@ -261,11 +261,19 @@ function get_option( $option, $default_value = false ) {
 function wp_prime_option_caches( $options ) {
 	$alloptions     = wp_load_alloptions();
 	$cached_options = wp_cache_get_multiple( $options, 'options' );
```

Comment:
There's no tests to ensure `$options` is an array of scalars, should there be?

Each value may need a check similar to the one at the start of the `*_options` functions too

https://github.com/WordPress/wordpress-develop/blob/6fb60e44e27e1b218ad71980f6de9bb7c5dd4418/src/wp-includes/option.php#L716-L722

Reply:
@peterwilsoncc I don't think either of this is necessary, here's why:
* Regarding the `is_scalar && trim` bit, I can't imagine anyone passing untrimmed option names to any of the functions. Maybe that was a real concern at some point, maybe it has always been only a theoretical concern. But since this is a new function, I don't see value in it. Otherwise we may as well include this trimming in literally _any_ core function that accepts a string identifier.
* The `wp_protect_special_option()` bit is only used in functions that write to the database so far. Given that, I don't think it's needed here (e.g. `get_option()` doesn't call it either).

Both of these are IMO theoretically nice to have, but I think there's too little value in including them to actually justify it. So I would vote no.